### PR TITLE
Change OpenTSDB queries to use POST(with request body) instead of GET(with query params)

### PR DESCRIPTION
**Pull request by bsura**

### CODE REVIEW

**Comment by bsura** [comment]
@rajsarkapally-sfdc @dilipdevaraj-sfdc Please review so we can push this out in our upcoming QA deployment.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/MetricQueryTransform.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+class MetricQueryTransform {
+	
+	private MetricQueryTransform() {}
+	
+	static class Serizlizer extends JsonSerializer<MetricQuery> {
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/MetricQueryTransform.java]: Correct class name spelling

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/MetricQueryTransform.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package com.salesforce.dva.argus.service.tsdb;
+
+import java.io.IOException;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/MetricQueryTransform.java]: Can you add some Javadoc on class and why it is needed?

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/DefaultTSDBService.java [hunk 3]
```diff
@@ -274,12 +274,11 @@ public void putMetrics(List<Metric> metrics) {
         Map<MetricQuery, List<Metric>> metricsMap = new HashMap<>();
         Map<MetricQuery, Future<List<Metric>>> futures = new HashMap<>();
         Map<MetricQuery, Long> queryStartExecutionTime = new HashMap<>();
-        String pattern = _readEndpoint + "/api/query?no_annotations=true&{0}";
+        String requestUrl = _readEndpoint + "/api/query";
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/DefaultTSDBService.java]: Is the no_annotations query parameter not needed now, or is it not supported in POST request to TSDB ?

Comment (bsura) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/DefaultTSDBService.java]: No more query params. Everything is added to the request body.

jgen.writeBooleanField("noAnnotations", true);

**Pull request merged**